app/services/wallet_service.rb

```
@@ -31,7 +31,8 @@ def spread_deposit(deposit)
         # NOTE: Consider min_collection_amount is defined per wallet.
         #       For now min_collection_amount is currency config.
         { address:                 w.address,
-          balance:                 w.current_balance(deposit.currency),
+          balance:                 w.current_balance(deposit.currency) * deposit.currency.price.to_d,
```

Let's raise an exception if the price is unknown.